pkg/cosigner/image_signer.go

```diff
@@ -0,0 +1,107 @@
+package cosigner
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/sigstore/cosign/cmd/cosign/cli"
+)
+
+type SignFunc func(ctx context.Context, ko cli.KeyOpts, annotations map[string]interface{}, imageRef string, certPath string, upload bool, payloadPath string, force bool, recursive bool) error
+
+type ImageSigner struct {
+	Logger   *log.Logger
+	signFunc SignFunc
+}
+
+const (
+	cosignRepositoryEnv       = "COSIGN_REPOSITORY"
+	cosignDockerMediaTypesEnv = "COSIGN_DOCKER_MEDIA_TYPES"
+)
+
+var (
+	secretLocation = "/var/build-secrets/cosign"
+)
+
+func NewImageSigner(logger *log.Logger, signFunc SignFunc) *ImageSigner {
+	return &ImageSigner{
+		Logger:   logger,
+		signFunc: signFunc,
+	}
+}
+
+func (s *ImageSigner) Sign(report lifecycle.ExportReport, annotations map[string]interface{}, cosignRepositories map[string]interface{}, cosignDockerMediaTypes map[string]interface{}) error {
+	if len(report.Image.Tags) < 1 {
+		s.Logger.Println("no image tag to sign")
```

Comment:
Do we actually think this would occur? It seems like an error case?

Reply:
Realistically, probably not going to occur. I will remove this